homeassistant/components/conversation/default_agent.py

```diff
@@ -384,6 +403,9 @@ def _make_names_list(self) -> TextSlotList:
                     # Skip configuration/diagnostic entities
                     continue
 
+                if entry.name:
```

Comment:
Without checking `has_entity_name` and falling back to `original_name`, you cannot assume `entry.name` is the name of the entity (it can even be None)

Reply:
You're already adding `state.name` so why would this be needed?